modules/host/negotiatedownload.go

```diff
@@ -157,22 +158,42 @@ func verifyPaymentRevision(existingRevision, paymentRevision types.FileContractR
 		return errLateRevision
 	}
 
-	// The new revenue comes out of the renter's valid outputs.
-	if paymentRevision.NewValidProofOutputs[0].Value.Add(expectedTransfer).Cmp(existingRevision.NewValidProofOutputs[0].Value) > 0 {
-		return errHighRenterValidOutput
+	// Determine the amount that was transfered from the renter.
+	if paymentRevision.NewValidProofOutputs[0].Value.Cmp(existingRevision.NewValidProofOutputs[0].Value) > 0 {
+		return extendErr("renter increased its valid proof output: ", errHighRenterValidOutput)
 	}
-	// The new revenue goes into the host's valid outputs.
-	if existingRevision.NewValidProofOutputs[1].Value.Add(expectedTransfer).Cmp(paymentRevision.NewValidProofOutputs[1].Value) < 0 {
-		return errLowHostValidOutput
+	fromRenter := existingRevision.NewValidProofOutputs[0].Value.Sub(paymentRevision.NewValidProofOutputs[0].Value)
+	// Verify that enough money was transfered.
+	if fromRenter.Cmp(expectedTransfer) < 0 {
```

Comment:
I don't know why I was so stubborn about it before. Clearly the code was hard to think about, as we both reviewed it and okay'd it but it was wrong. Should have made the code more intuitive from the start.